src/file_operations.rs

```diff
@@ -86,6 +89,14 @@ impl FileOperations {
         self.operations.push(Op::MkDirAll(self.root.join(name)))
     }
 
+    pub fn link<P: AsRef<Path>>(&mut self, source: P, dest: P) {
```

Comment:
I don't think so.  The `AsRef<Path>` itself will convert anything that has implemented the `AsRef` trait for `Path`, which includes `String`, `&str`, `Path` and `PathBuf`.  I think it's equally valid to pass in any combination of these things, and an unnecessary headache to need to convert them explicitly.